### PR TITLE
Adds 404/500 support via pass-thru to Angular

### PR DESCRIPTION
**Pull request by kwhitley**
### Notes
- `/abc` will be redirected to `/#/abc`.  This attempts to let angular pick up the route, and if failing, pass along to its `errorController` as a 404.
- `/fail` is a specific (test) error route that triggers a server error, that is then caught by the server side error handler.  The error handler redirects the error to `/#/error/:url`, which is also handled by `errorController`, but with a `server-error.html` template

### CODE REVIEW

File: src/server/lib/errors/index.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+var express = require('express');
+var app     = module.exports = express();
+
+app.get('/fail', function(req, res, next) {
```
Comment (kwhitley) [inline src/server/lib/errors/index.js]: This is for testing purposes only and will be removed

**Comment by ericclemmons** [comment]
I'll be at the dr. this afternoon. Screenshot would work? 

## 

Eric Clemmons
Sent with Sparrow (http://www.sparrowmailapp.com/?sig)

**Comment by kwhitley** [comment]
**GET** `localhost:3000/fail?something`

Redirects to...
Snapshot 1:22:13 3:22 PM

**Comment by kwhitley** [comment]
### Error: 404

Snapshot 1:22:13 5:09 PM

### Error: 500

Snapshot 1:22:13 5:11 PM

### Error: Unknown

Snapshot 1:22:13 5:12 PM

**Comment by kwhitley** [comment]
:tada:

**Pull request merged**